src/pages/Profile.js

```diff
@@ -1,22 +1,59 @@
-import React from 'react';
-import { Link } from 'react-router-dom';
+import React, {useContext, useEffect, useState} from 'react';
+import {Link} from 'react-router-dom';
+import axios from "axios";
+import {AuthContext} from "../context/AuthContext";
 
 function Profile() {
-  return (
-    <>
-      <h1>Profielpagina</h1>
-      <section>
-        <h2>Gegevens</h2>
-        <p><strong>Gebruikersnaam:</strong> hardcoded-test</p>
-        <p><strong>Email:</strong> hardcoded@test.com</p>
-      </section>
-      <section>
-        <h2>Strikt geheime profiel-content</h2>
-        <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ab alias cum debitis dolor dolore fuga id molestias qui quo unde?</p>
-      </section>
-      <p>Terug naar de <Link to="/">Homepagina</Link></p>
-    </>
-  );
+
+    const {isAuth, user} = useContext(AuthContext);
+    const token = localStorage.getItem('token');
+    const [privateContent, setPrivateContent] = useState({});
+
+    useEffect(() => {
+        const controller = new AbortController();
+        async function getPrivateContent() {
+            try {
+                const response = await axios.get('http://localhost:3000/660/private-content', {
+                    headers: {
+                        "Content-Type": "application/json",
+                        Authorization: `Bearer ${token}`,
+                    },
+                    signal: controller.signal,
+                })
+                console.log(response);
+                setPrivateContent(response.data);
+            } catch (e) {
+                console.error(e)
+            }
+        }
+        void getPrivateContent();
+        return function cleanup() {
+            controller.abort();
+        }
+    }, []);
+
+
```

Comment:
Je zou dan hier die check kunnen doen
```suggestion
if (privateContent === undefined) return null; // of <p>Loading</p>
```